Modules/pyexpat.c

```diff
@@ -1880,10 +1936,37 @@ MODULE_INITFUNC(void)
 #endif
 
     /* export using capsule */
-    capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
-    if (capi_object)
-        PyModule_AddObject(m, "expat_CAPI", capi_object);
-    return m;
+    PyObject *capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
+    if (capi_object == NULL) {
+        return -1;
+    }
+
+    if (PyModule_AddObject(mod, "expat_CAPI", capi_object) < 0) {
+        Py_DECREF(capi_object);
+        return -1;
+    }
+
+    return 0;
+}
+
+static PyModuleDef_Slot pyexpat_slots[] = {
+    {Py_mod_exec, pyexpat_exec},
+    {0, NULL}
+};
+
+static struct PyModuleDef pyexpatmodule = {
+    PyModuleDef_HEAD_INIT,
+    .m_name = MODULE_NAME,
+    .m_doc = pyexpat_module_documentation,
+    .m_size = 0,
+    .m_methods = pyexpat_methods,
+    .m_slots = pyexpat_slots
+};
+
+PyMODINIT_FUNC
+PyInit_pyexpat(void)
+{
+    return PyModuleDef_Init(&pyexpatmodule);
```

Comment:
I would prefer to convert the pyexpat module to multi-phase init in a separated PR. For example, I would prefer to first see a second PR to convert static types to heap types.

Reply:
@vstinner I misunderstood your previous comment:
>Since you plan to convert pyexpat to multiphase init, could you already add an "exec" function, and modify the PyInit to call it? So later you will have no replace all "goto error" with "return -1".

I'll remove multi-phase init